performance/reports/rp.py

```diff
@@ -59,55 +61,73 @@ def generate_weekly_report_for_date(date_start, report_output_path):
 
-        signup_attempts = piwik.get_all_signup_attempts_for_rp(rp, date_start)
+def generate_weekly_report_df(date_start, df_verifications_by_rp):
+    df = add_piwik_data(date_start, df_verifications_by_rp)
+    transform_metrics(df)
+    return df
 
-        single_idp_attempts = piwik.get_all_single_idp_attempts_for_rp(rp, date_start)
 
-        # TODO if the RP is not found (e.g. due to no successful signins) then we may need to add a row?
-        df_all.loc[(df_all['rp'] == rp), 'all_referrals'] = all_referrals
-        df_all.loc[(df_all['rp'] == rp), 'signin_attempt'] = signin_attempts
-        df_all.loc[(df_all['rp'] == rp), 'signup_attempt'] = signup_attempts
-        df_all.loc[(df_all['rp'] == rp), 'single_idp_attempt'] = single_idp_attempts
-    df_all['signin_rate (deprecated)'] = df_all['signin_success'] / df_all['signin_attempt']
-    df_all['signup_rate (deprecated)'] = df_all['signup_success'] / df_all['signup_attempt']
-    df_all['all_referrals_with_intent'] = df_all['signin_attempt'] + df_all['signup_attempt'] + df_all[
+def transform_metrics(df):
+    # df['signin_rate (deprecated)'] = df['signin_success'] / df['signin_attempt']
+    # df['signup_rate (deprecated)'] = df['signup_success'] / df['signup_attempt']
+    df['all_referrals_with_intent'] = df['signin_attempt'] + df['signup_attempt'] + df[
         'single_idp_attempt']
-    df_all['success'] = df_all['signin_success'] + df_all['signup_success']
-    df_all['success_rate'] = df_all['success'] / df_all['all_referrals_with_intent']
-    df_all['success_fraction_signup'] = df_all['signup_success'] / df_all['success']
-    df_all['success_fraction_signin'] = df_all['signin_success'] / df_all['success']
-    df_all['success_signup_rate'] = df_all['signup_success'] / df_all['all_referrals_with_intent']
-    df_all['success_signin_rate'] = df_all['signin_success'] / df_all['all_referrals_with_intent']
-    df_all['dropout'] = df_all['all_referrals_with_intent'] - df_all['signin_success'] - df_all['signup_success']
-    df_all['dropout_rate'] = df_all['dropout'] / df_all['all_referrals_with_intent']
-
+    df['success'] = df['signin_success'] + df['signup_success']
+    df['success_fraction_signup'] = df['signup_success'] / df['success']
+    df['success_fraction_signin'] = df['signin_success'] / df['success']
+    # df['success_rate'] = df['success'] / df['all_referrals_with_intent']
+    # df['success_signup_rate'] = df['signup_success'] / df['all_referrals_with_intent']
+    # df['success_signin_rate'] = df['signin_success'] / df['all_referrals_with_intent']
+    # df['dropout'] = df['all_referrals_with_intent'] - df['signin_success'] - df['signup_success']
+    # df['dropout_rate'] = df['dropout'] / df['all_referrals_with_intent']
     # ### Ineligible (will not work / might not work) - with journey type segment
     # upvson 'GOV.UK Verify will not work for you' page, segmented by 'registration' custom variable
```

Comment:
I don't really follow these comments, but it looks like they aren't relevant to the code now that the `get_visits_will_not_work` and `get_visits_might_not_work` calls have moved?